pkg/reconciler/instances/connectivityproxy/rendering/chartprovider.go

```diff
@@ -0,0 +1,63 @@
+package rendering
+
+import (
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	internalKubernetes "github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type FilterFunc func([]*unstructured.Unstructured) ([]*unstructured.Unstructured, error)
+
+func NewProviderWithFilters(provider chart.Provider, filterFuncs ...FilterFunc) chart.Provider {
+	return provider.WithFilter(func(manifest string) (string, error) {
+		if len(filterFuncs) == 0 {
+			return manifest, nil
+		}
+
+		unstructs, err := internalKubernetes.ToUnstructured([]byte(manifest), true)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to convert manifest to unstructured object")
+		}
+
+		for _, filterFunc := range filterFuncs {
+			if filterFunc != nil {
+				unstructs, err = filterFunc(unstructs)
+				if err != nil {
+					return "", err
+				}
+			}
+		}
+
+		newManifest, err := serialize(unstructs)
+		if err != nil {
+			return "", err
+		}
+
+		return newManifest, nil
+	})
+}
+
```

Comment:
Maybe it is worthy to move that function to `kubernetes` package besides  `kubernetes.ToUnstructured` method for common benefit and call it `kubernetes.FromUnstructured`.

Reply:
After discussion - no action will be done.